execute/report/report.go

```diff
@@ -430,6 +487,10 @@ func (b *execReportBuilder) buildSingleChainReport(
 	}
 
 	// Unless there is a message limit, attempt to build a report for executing all ready messages.
+	// It is possible that the report produced here is invalid for some reason, such as
+	// report size or gas usage.
+	// In that case, we will execute the loop below to iteratively build a report
+	// with fewer messages until we find a valid report.
```

Comment:
I'm not sure why we have these as separate loops and why we have this `if` condition. It seems to me that the loop below will eventually fit everything into one report if it can. Is this an optimization @winder ?